File: csharp/DVSClient/DVSClient/server/address/datasets/RestApiGetDatasetsResponse.cs
using Newtonsoft.Json;

namespace DVSClient.Server.Address.Datasets
{
    public class RestApiGetDatasetsResponse : RestApiResponse
    {
        [JsonProperty("error")]
        public RestApiResponseError? Error { get; set; }

        [JsonProperty("result")]
        public IEnumerable<RestApiAddressDatasetResult>? Result { get; set; }
    }
}

File: typescript/src/server/address/datasets/restApiGetDatasetsResponse.ts
import { RestApiResponse } from "../../restApiResponse";
import { RestApiResponseError } from "../../restApiResponseError";
import { RestApiAddressDatasetResult } from "./restApiAddressDatasetResult";

export type RestApiGetDatasetsResponse = RestApiResponse & {
    error?: RestApiResponseError;
    result?: RestApiAddressDatasetResult[];
}
